src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,216 @@
+                // epsilon
+                real_type delta = this->impact_parameter(epsilon);
+                CHECK(delta <= delta_max && delta >= delta_min);
+                // Calculate g1 "rejection" function
+                reject_threshold
+                    = celeritas::max(this->screening_phi1_aux(delta), 0.0)
+                      / celeritas::max(this->screening_phi1_aux(delta_min),
+                                       0.0);
+                CHECK(reject_threshold > 0.0 && reject_threshold <= 1.0);
+            }
+            else
+            {
+                // Used to sample from f2
+                epsilon = epsilon_min
+                          + (0.5 - epsilon_min) * generate_canonical(rng);
+                CHECK(epsilon >= epsilon_min && epsilon <= 0.5);
+                // Calculate delta given the element atomic number and sampled
+                // epsilon
+                real_type delta = this->impact_parameter(epsilon);
+                CHECK(delta <= delta_max && delta >= delta_min);
+                // Calculate g2 "rejection" function
+                reject_threshold
+                    = celeritas::max(this->screening_phi2_aux(delta), 0.0)
+                      / celeritas::max(this->screening_phi2_aux(delta_min),
+                                       0.0);
+                CHECK(reject_threshold > 0.0 && reject_threshold <= 1.0);
+            }
+        } while (BernoulliDistribution(1.0 - reject_threshold)(rng));
+    }
+
+    // Select charges randomly
+    real_type total_energy = inc_energy_.value()
+                             + 2.0 / shared_.inv_electron_mass;
+    secondaries[0].energy = units::MevEnergy{(1.0 - epsilon) * total_energy
+                                             - 1 / shared_.inv_electron_mass};
+    secondaries[1].energy = units::MevEnergy{epsilon * total_energy
+                                             - 1 / shared_.inv_electron_mass};
+    if (BernoulliDistribution(0.5)(rng))
+    {
+        swap2<units::MevEnergy>(secondaries[0].energy, secondaries[1].energy);
+    }
+
+    // Sample pair directions
+    UniformRealDistribution<real_type> sample_phi(0.0, 2.0 * constants::pi);
+    for (int i = 0; i < 2; ++i)
+    {
+        real_type umax
+            = 2
+              * (1.0
+                 + secondaries[i].energy.value() * shared_.inv_electron_mass);
+        real_type u;
+        do
+        {
+            real_type uu
+                = -std::log(generate_canonical(rng) * generate_canonical(rng));
+            u = BernoulliDistribution(0.25)(rng) ? uu * 1.6
+                                                 : uu * (1.0 / 1.875);
+        } while (u > umax);
+
+        real_type cost = (i == 0 ? -1.0 : 1.0)
+                         * (1.0 - 2.0 * ipow<2>(u / umax));
+        secondaries[i].direction
+            = rotate(from_spherical(cost, sample_phi(rng)), inc_direction_);
+        normalize_direction(&secondaries[i].direction);
+    }
```

Comment:
Unsure about this.

The unit tests are fine except for the `dot_product` since the directions/momenta are wrong.

Reply:
Yeah, I think the issue is that as written the two directions are independent but they need to be correlated. I'll take a crack at porting over the "modified tsai" angle sampler.

Reply:
I have followed the `G4ModifiedTsai` sampler in more recent code. I will push shortly to show it (just committing to another project).